### PR TITLE
Add APM/Kibana association e2e test

### LINKED ISSUE #3154: Add e2e tests for APM / Kibana association
Following-up #3043 we should add some end to end tests to validate that an APM Server is correctly associated with Kibana.

### PR DESCRIPTION
**Pull request by barkbay**
Add an e2e test for the APM/Kibana association. The test scenario is the one described [here](https://github.com/elastic/cloud-on-k8s/issues/3154#issuecomment-635360186).

Fix #3154

### CODE REVIEW

File: test/e2e/apm/association_test.go [hunk 1]
```diff
@@ -43,6 +45,41 @@ func TestCrossNSAssociation(t *testing.T) {
...
+
+	apmBuilder := apmserver.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithKibanaRef(kbBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext().
+		WithConfig(map[string]interface{}{
+			"apm-server.ilm.enabled":                           false,
+			"setup.template.settings.index.number_of_replicas": 0, // avoid ES yellow state on a 1 node ES cluster
```
Comment (sebgl) [inline test/e2e/apm/association_test.go:77]: :o I'm surprised APM does not rely on `auto_expand_replicas: "0-1"` similar to what Kibana is doing.

File: test/e2e/test/apmserver/checks_apm.go [hunk 2]
```diff
@@ -264,3 +267,53 @@ func countIndex(esClient client.Client, indexName string) (int, error) {
...
 	return countResult.Count, nil
 }
+
+const sampleDefaultAgentConfiguration = `{"service":{},"settings":{"transaction_sample_rate":"1","capture_body":"errors","transaction_max_spans":"99"}}`
+
+// CheckAgentConfiguration creates an agent configuration through Kibana and then check that the APM Server is able to retrieve it.
+func (c *apmClusterChecks) CheckAgentConfiguration(apm apmv1.ApmServer, k *test.K8sClient) test.StepList {
+	apmVersion := version.MustParse(apm.Spec.Version)
+
+	if !(apmVersion.IsSameOrAfter(apmv1.ApmAgentConfigurationMinVersion) && apm.Spec.KibanaRef.IsDefined()) {
```
Comment (sebgl) [inline test/e2e/test/apmserver/checks_apm.go:277]: isn't the check on `apm.Spec.KibanaRef.IsDefined() ` good enough?

File: test/e2e/apm/association_test.go (hunk 1 above)
Comment (pebrc) [reply test/e2e/apm/association_test.go:77]: Looks like they considered it at some point https://github.com/elastic/apm-server/pull/503

File: test/e2e/test/apmserver/checks_apm.go [hunk 3]
```diff
@@ -264,3 +267,53 @@ func countIndex(esClient client.Client, indexName string) (int, error) {
...
+				assert.NoError(t, k.Client.Get(apm.Spec.KibanaRef.WithDefaultNamespace(apm.Namespace).NamespacedName(), &kb))
+
+				password, err := k.GetElasticPassword(apm.Spec.ElasticsearchRef.WithDefaultNamespace(apm.Namespace).NamespacedName())
+				assert.NoError(t, err)
+
+				uri := "/api/apm/settings/agent-configuration"
+
+				// URI is slightly different before 7.7.0, we need to add "/new" at the end
+				if !apmVersion.IsSameOrAfter(version.MustParse("7.7.0")) {
+					uri += "/new"
```
Comment (thbkrkr) [inline test/e2e/test/apmserver/checks_apm.go:295]: I understand we add the `/new` prefix when the version of the APM Server is equal or greater than 7.7.0. Is it not the opposite that we want?

File: test/e2e/test/apmserver/checks_apm.go (hunk 2 above)
Comment (barkbay) [reply test/e2e/test/apmserver/checks_apm.go]: Yes, I have been a bit paranoid on this one 😄 Let's remove it and see how it goes.

**Pull request merged**